runtimes/web/src/ui/app.ts

```diff
@@ -14,6 +14,7 @@ import { Notifications } from "./notifications";
 
 class InputState {
     gamepad = [0, 0, 0, 0];
+    gamepadUnavailableWarned = new Set<number>;
```

Comment:
What do you think about keying this off the gamepad's ID instead of the index?

I guess that would allow someone to test plugging and unplugging multiple different types of gamepads one at a time and seeing the error message for each invalid one, even if the index is the same.

Reply:
Done.

Though isn't browser supposed to give different gamepads different indices? At least that's how I understood MDN. I guess that it's better this way anyway since the code is a bit simpler.

Reply:
From what I understand the index is simply the order that the gamepad was connected, but different gamepads can have the same index if you unplug and replug in a different gamepad.